mteb/tasks/Retrieval/multilingual/MLQARetrieval.py

```diff
@@ -0,0 +1,154 @@
+    "mlqa.es.vi": ["spa-Latn", "vie-Latn"],
+    "mlqa.es.zh": ["spa-Latn", "zho-Hans"],
+    "mlqa.hi.ar": ["hin-Deva", "ara-Arab"],
+    "mlqa.hi.de": ["hin-Deva", "deu-Latn"],
+    "mlqa.hi.en": ["hin-Deva", "eng-Latn"],
+    "mlqa.hi.es": ["hin-Deva", "spa-Latn"],
+    "mlqa.hi.hi": ["hin-Deva", "hin-Deva"],
+    "mlqa.hi.vi": ["hin-Deva", "vie-Latn"],
+    "mlqa.hi.zh": ["hin-Deva", "zho-Hans"],
+    "mlqa.vi.ar": ["vie-Latn", "ara-Arab"],
+    "mlqa.vi.de": ["vie-Latn", "deu-Latn"],
+    "mlqa.vi.en": ["vie-Latn", "eng-Latn"],
+    "mlqa.vi.es": ["vie-Latn", "spa-Latn"],
+    "mlqa.vi.hi": ["vie-Latn", "hin-Deva"],
+    "mlqa.vi.vi": ["vie-Latn", "vie-Latn"],
+    "mlqa.vi.zh": ["vie-Latn", "zho-Hans"],
+    "mlqa.zh.ar": ["zho-Hans", "ara-Arab"],
+    "mlqa.zh.de": ["zho-Hans", "deu-Latn"],
+    "mlqa.zh.en": ["zho-Hans", "eng-Latn"],
+    "mlqa.zh.es": ["zho-Hans", "spa-Latn"],
+    "mlqa.zh.hi": ["zho-Hans", "hin-Deva"],
+    "mlqa.zh.vi": ["zho-Hans", "vie-Latn"],
+    "mlqa.zh.zh": ["zho-Hans", "zho-Hans"],
+}
+
+
+def extend_lang_pairs() -> dict[str, list[str]]:
+    eval_langs = {}
+    for langs in _LANGUAGES.values():
+        lang_pair = langs[0] + "_" + langs[1]
+        eval_langs[lang_pair] = langs
+    return eval_langs
+
+
+_EVAL_LANGS = extend_lang_pairs()
+
+
+class MLQARetrieval(AbsTaskRetrieval, CrosslingualTask):
+    metadata = TaskMetadata(
+        name="MLQARetrieval",
+        description="""MLQA (MultiLingual Question Answering) is a benchmark dataset for evaluating cross-lingual question answering performance.
+        MLQA consists of over 5K extractive QA instances (12K in English) in SQuAD format in seven languages - English, Arabic,
+        German, Spanish, Hindi, Vietnamese and Simplified Chinese. MLQA is highly parallel, with QA instances parallel between
+        4 different languages on average.""",
+        reference="https://huggingface.co/datasets/mlqa",
+        dataset={
+            "path": "mlqa",
+            "revision": "397ed406c1a7902140303e7faf60fff35b58d285",
+        },
+        type="Retrieval",
+        category="s2p",
+        eval_splits=["validation", "test"],
+        eval_langs=_EVAL_LANGS,
+        main_score="ndcg_at_10",
+        date=("2019-01-01", "2020-12-31"),
+        form=["written"],
+        domains=["Encyclopaedic"],
+        task_subtypes=["Question answering"],
+        license="cc-by-sa-3.0",
+        socioeconomic_status="mixed",
+        annotations_creators="human-annotated",
+        dialect=[],
+        text_creation="found",
+        bibtex_citation="""@article{lewis2019mlqa,
+        title = {MLQA: Evaluating Cross-lingual Extractive Question Answering},
+        author = {Lewis, Patrick and Oguz, Barlas and Rinott, Ruty and Riedel, Sebastian and Schwenk, Holger},
+        journal = {arXiv preprint arXiv:1910.07475},
+        year = 2019,
+        eid = {arXiv: 1910.07475}
+        }""",
+        n_samples={"test": 158083, "validation": 15747},
```

Comment:
Yeah for now the number of samples for multilingual and crosslingual tasks is just equal to the sum of samples accross all languages and pairs. We're still thinking about how to implement a faster version of tasks for Retrieval, just as we did in [Clustering,](https://github.com/embeddings-benchmark/mteb/blob/main/mteb/abstasks/AbsTaskClusteringFast.py) do you have any suggestions? We can't simply subsample imo, because it will impact the ndcg score..

Reply:
I agree, downsampling is quite bad, and the large amount of samples issue itself in this pull request spawns from the amount of language pairs, which is natural and unavaoidable. 

A non trivial solution would be very nice, but I would hope for a proper reasoning defending that choice.

For now, I would keep as is (since the embeddings are currently not calculated multiple times, so all good). I would still suggest to avoid the validation set, test set is enough.